src/psyclone/generator.py

```diff
@@ -139,32 +132,24 @@ def handle_script(script_name, info, function_name, is_optional=False):
         # already exist elsewhere in the system path
         sys_path_prepended = True
         sys.path.insert(0, filepath)
-        try:
-            transmod = __import__(module_name)
-        except Exception as error:
-            raise GenerationError(
-                f"generator: attempted to import specified PSyclone "
-                f"transformation module '{module_name}' but a problem was "
-                f"found: {error}") from error
-        if callable(getattr(transmod, function_name, None)):
-            try:
-                func_call = getattr(transmod, function_name)
-                info = func_call(info)
-            except Exception:
-                exc_type, exc_value, exc_traceback = sys.exc_info()
-                lines = traceback.format_exception(exc_type, exc_value,
-                                                   exc_traceback)
-                e_str = '{\n' + ''.join('    ' + ln for ln in lines[2:]) + '}'
-                # pylint: disable=raise-missing-from
-                raise GenerationError(
-                    f"generator: specified PSyclone transformation module "
-                    f"'{module_name}'\nraised the following exception during "
-                    f"execution...\n{e_str}\nplease check your script")
-        elif not is_optional:
-            raise GenerationError(
-                f"generator: attempted to use specified PSyclone "
-                f"transformation module '{module_name}' but it does not "
-                f"contain a '{function_name}' function")
+        recipe_module = importlib.import_module(module_name)
```

Comment:
I should have mentioned this on the PR description but forgot, I deleted the previous try/except in purpose because I never liked that we catched and refactor script errors. I think recipe writers are Python developers and should expect the proper Python error in the script if they make a mistake. Now they get the untrapped exception:
```
$ psyclone -s error.py test.f90
Traceback (most recent call last):
  File "/home/sergi/workspace/psyclone/.venv/bin/psyclone", line 42, in <module>
    main(sys.argv[1:])
  File "/home/sergi/workspace/psyclone/PSyclone/src/psyclone/generator.py", line 548, in main
    code_transformation_mode(input_file=args.filename,
  File "/home/sergi/workspace/psyclone/PSyclone/src/psyclone/generator.py", line 707, in code_transformation_mode
    transformation_recipe, files_to_skip = load_script(recipe_file)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sergi/workspace/psyclone/PSyclone/src/psyclone/generator.py", line 136, in load_script
    recipe_module = importlib.import_module(module_name)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sergi/.local/share/mise/installs/python/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/home/sergi/workspace/psyclone/PSyclone/error.py", line 1, in <module>
    invalid
NameError: name 'invalid' is not defined
```

I think the bottom is very readable (and there are some IDEs that use this to provide hints and fixes), previously we put everything in a GeneratorError (breaking IDEs assistance) and make things harder because we had to read upwards to see the real issue.

Reply:
Yes, that is a lot nicer. Thanks.